Dockerfile

```diff
@@ -10,16 +10,31 @@ RUN apk add python3 py3-pip git \
 CMD git ls-files --cached -z -- '*.rst' \
     | xargs -0 -- python3 -m sphinxlint --enable all --disable trailing-whitespace --max-line-length 2000
 
-FROM php:${PHP_VERSION}-cli-alpine${ALPINE_VERSION} as php
+FROM php:${PHP_VERSION}-cli-alpine${ALPINE_VERSION} as base
+# https://blog.codito.dev/2022/11/composer-binary-only-docker-images/
+# https://github.com/composer/docker/pull/250
+COPY --from=composer/composer:2-bin /composer /usr/local/bin/composer
 
+FROM base as vendor
+COPY composer.json /fixer/composer.json
+WORKDIR /fixer
+RUN composer install --prefer-dist --no-dev --optimize-autoloader --no-scripts
```

Comment:
i would wonder to reuse phar file here, but non-blocking ;)

Reply:
Yo dawg! We encapsulated your encapsulated code, so you can exec while you exec 😉.

![image](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/1d7f8c88-c026-4cd5-acd6-d0aa341408fb)

But seriously, not worth it. We wrap the code with Docker image, so it's already encapsulated and we don't need another layer of encapsulation. Fixer is installed in separate dir, there's no dependency conflict with the code which is mounted and is about to be analysed. For debugging purposes it's much better if Fixer's code is openly available within the image.